File: pset1/credit.c
#include <cs50.h>
#include <stdio.h>

int checksum(long num)
{
    long count = 1, result, sum1 = 0, sum2 = 0, total = 0;
    long number = num * 10, digit;
    while (number != 0)
    {
        count = number / 100;
        number /= 100;
        digit = count % 10;
        printf("%lu", digit);
        sum1 += (digit * 2) / 10 + (digit * 2) % 10;
    }
    printf("Sum1: %lu", sum1);
    number = num;
    sum2 = number % 10;
    count = 1;
    while (number != 0)
    {
        count = number / 100;
        number /= 100;
        digit = count % 10;
        sum2 += digit;
    }
    printf("Sum2: %lu", sum2);
    total = sum1 + sum2;
    return total;
}

int main()
{
    long number;
    long final;
    do
    {
        number = get_long("Number: ");
    }
    while (number < 0);
    final = checksum(number);
    
    if (number / 10000000000000 == 34 || number / 10000000000000 == 37)
    {

    }
    else if (number / 100000000000000 == 51 || number / 100000000000000 == 52 || number / 100000000000000 == 53 || number / 100000000000000 == 54 || number / 100000000000000 == 55)
    {
        //MasterCard
    }
    else if (number / 100000000000 == 4 || number / 100000000000000 == 4)
    {
        //Visa
    }
    else
    {
        printf("INVALID");
    }
}

File: pset6/credit/credit.py
from cs50 import get_int

number = get_int("Card Number: ")
number = str(number)
possible = 'INVALID'


def checksum(number):
    digits = list(map(int, str(number)))
    odd_sum = sum(digits[-1::-2])
    sum_val = 0
    for digit in digits[-2::-2]:
        digit *= 2
        if digit > 9:
            digit = (digit % 10) + 1
        sum_val += digit
    if (odd_sum + sum_val) % 10 == 0:
        return True
    else:
        return False;


if len(number) == 13:
    if number[:1] == "4":
        possible = 'VISA'
if len(number) == 15:
    if number[:2] in ["34", "37"]:
        possible = 'AMEX'
if len(number) == 16:
    if number[:1] == "4":
        possible = 'VISA'
    elif number[:2] in ["51", "52", "53", "54", "55"]:
        possible = 'MASTERCARD'
if possible != 'INVALID':
    if checksum(number) == True:
        print(possible+'\n')
    else:
        print('INVALID\n')
else:
    print('INVALID\n')


